native/src/main/java/io/ballerina/stdlib/serdes/SchemaGenerator.java

```diff
@@ -43,201 +75,538 @@
+            String proto3 = (String) serdes.getNativeData(PROTO3);
+            file.write(proto3);
+        } catch (IOException e) {
+            String errorMessage = FAILED_WRITE_FILE + e.getMessage();
+            return createSerdesError(errorMessage, SERDES_ERROR);
         }
-        serdes.addNativeData(SCHEMA_NAME, schema);
         return null;
     }
 
-    private static ProtobufMessage buildProtobufMessageFromBallerinaTypedesc(BTypedesc typedesc) {
-        Type type = typedesc.getDescribingType();
-
-        if (type.getTag() <= TypeTags.BOOLEAN_TAG) {
-            String ballerinaToProtoMap = DataTypeMapper.getProtoTypeFromTag(type.getTag());
-            ProtobufMessageBuilder messageBuilder = ProtobufMessage.newMessageBuilder(ballerinaToProtoMap);
-            buildProtobufMessageForBallerinaPrimitiveType(messageBuilder, ballerinaToProtoMap, ATOMIC_FIELD_NAME, 1);
-            return messageBuilder.build();
-        } else if (type.getTag() == TypeTags.UNION_TAG) {
-            ProtobufMessage protobufMessage = buildProtobufMessageForBallerinaUnionType(type, UNION_FIELD_NAME);
-            ProtobufMessageBuilder messageBuilder = ProtobufMessage.newMessageBuilder(UNION_BUILDER_NAME);
-            messageBuilder.addNestedMessage(protobufMessage);
-            messageBuilder.addField(OPTIONAL_LABEL, UNION_FIELD_NAME, ATOMIC_FIELD_NAME, 1);
-            return messageBuilder.build();
-        } else if (type.getTag() == TypeTags.ARRAY_TAG) {
-            ArrayType arrayType = (ArrayType) type;
-            ProtobufMessageBuilder messageBuilder = ProtobufMessage.newMessageBuilder(ARRAY_BUILDER_NAME);
-            buildProtobufMessageForBallerinaArrayType(messageBuilder, arrayType, ARRAY_FIELD_NAME, 1, 1);
-            return messageBuilder.build();
-        } else if (type.getTag() == TypeTags.RECORD_TYPE_TAG) {
-            RecordType recordType = (RecordType) type;
-            return buildProtobufMessageForBallerinaRecordType(recordType.getFields(), type.getName());
-        } else {
-            throw createSerdesError(UNSUPPORTED_DATA_TYPE + type.getName(), SERDES_ERROR);
+    private static ProtobufMessageBuilder buildProtobufMessageFromBallerinaTypedesc(Type ballerinaType) {
+
+        ProtobufMessageBuilder messageBuilder;
+        String messageName;
+
+        switch (ballerinaType.getTag()) {
+            case TypeTags.INT_TAG:
+            case TypeTags.BYTE_TAG:
+            case TypeTags.FLOAT_TAG:
+            case TypeTags.STRING_TAG:
+            case TypeTags.BOOLEAN_TAG: {
+                int fieldNumber = 1;
+                messageName = Utils.createMessageName(ballerinaType.getName());
+                messageBuilder = new ProtobufMessageBuilder(messageName);
+                generateMessageDefinitionForPrimitiveType(
+                        messageBuilder,
+                        ballerinaType,
+                        ATOMIC_FIELD_NAME,
+                        fieldNumber);
```

Comment:
Let's not use multiple lines for parameters. If the line exceeds the limit, break it into two lines. Check similar places.

Reply:
Updated all the places